static/src/js/components/OrderStatus/OrderStatusItem.js

```diff
@@ -1,51 +1,93 @@
-import React from 'react'
+import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
 
-import { getStateFromOrderStatus } from '../../util/orderStatus'
+import { getStateFromOrderStatus, aggregatedOrderStatus } from '../../util/orderStatus'
 import OrderStatusItemBody from './OrderStatusItemBody'
 
 import './OrderStatusItem.scss'
 
-export const OrderStatusItem = ({ collection, onChangePath, type }) => {
-  const { collection_metadata: collectionMetadata, access_method: accessMethod } = collection
-  const { dataset_id: datasetId } = collectionMetadata
-  const { order = {} } = accessMethod
+export class OrderStatusItem extends PureComponent {
+  componentDidMount() {
+    const { collection, onFetchRetrievalCollection, type } = this.props
 
-  let {
-    order_status: orderStatus
-  } = order
+    if (collection && !['download', 'opendap'].includes(type)) {
+      const { id } = collection
 
-  if (type === 'download') orderStatus = 'complete'
+      if (id) {
+        // Fetch the retrieval collection before waiting for the interval
+        onFetchRetrievalCollection(id)
 
-  const className = classNames(
-    'order-status-item',
-    {
-      'order-status-item--success': getStateFromOrderStatus(orderStatus) === 'success',
-      'order-status-item--in-progress': getStateFromOrderStatus(orderStatus) === 'in progress',
-      'order-status-item--errored': getStateFromOrderStatus(orderStatus) === 'errored'
+        // Start refreshing the retrieval collection every 20 seconds
+        setInterval(() => {
+          try {
+            onFetchRetrievalCollection(id)
+          } catch (e) {
+            console.log(e)
+          }
+        }, 60000)
```

Comment:
Would it make sense to abstract these delay times out into the config somewhere?